s3/src/main/scala/geotrellis/spark/io/s3/S3LayerUpdater.scala

```diff
@@ -37,71 +38,40 @@ class S3LayerUpdater(
 ) extends LayerUpdater[LayerId] with LazyLogging {
 
   def rddWriter: S3RDDWriter = S3RDDWriter
+  def _rddWriter(): S3RDDWriter = rddWriter
+
+  class InnerS3LayerWriter(
+    attributeStore: AttributeStore,
+    bucket: String,
+    keyPrefix: String
+  ) extends S3LayerWriter(attributeStore, bucket, keyPrefix) {
+    override def rddWriter() = _rddWriter
+  }
+
+  val as = attributeStore.asInstanceOf[S3AttributeStore]
```

Comment:
This cast looks very dirty and makes `LayerUpdater` incompatible with other backends, it's a normal use case when our `AttributeStore` and the `catalog` itself are in different backends. What do you think about it? Or mb to make `AttributeStore` types more restrictive in `2.0`?

Reply:
Yes, the interface for `S3LayerUpdater` is not great. I'm going to make these patch fields private and call it a day. The updaters should be removed in 2.0 and the way they're constructed right now should never trigger the error case.